src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +34,19 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).orElseThrow(() ->
+                new RuntimeException("User with such email doesn`t exist"));
+        ShoppingCart cart = shoppingCartService.getByUser(user);
         return orderResponseDtoMapper.mapToDto(orderService.completeOrder(cart));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId))
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        return orderService.getOrdersHistory(userService
+                        .findByEmail(authentication.getName())
+                        .orElseThrow(() ->
+                                new RuntimeException("User with such email doesn`t exist")))
```

Comment:
It's better to replace this exception with NoSuchElementException. Fix the same issue in all places